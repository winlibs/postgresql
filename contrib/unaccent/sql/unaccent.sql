CREATE EXTENSION unaccent;

-- must have a UTF8 database
SELECT getdatabaseencoding();

SET client_encoding TO 'UTF8';

SELECT unaccent('foobar');
SELECT unaccent('ёлка');
SELECT unaccent('ЁЖИК');
SELECT unaccent('˃˖˗˜');
SELECT unaccent('À');  -- Remove combining diacritical 0x0300
SELECT unaccent('℃℉'); -- degree signs
SELECT unaccent('℗'); -- sound recording copyright

SELECT unaccent('unaccent', 'foobar');
SELECT unaccent('unaccent', 'ёлка');
SELECT unaccent('unaccent', 'ЁЖИК');
SELECT unaccent('unaccent', '˃˖˗˜');
SELECT unaccent('unaccent', 'À');
SELECT unaccent('unaccent', '℃℉');
SELECT unaccent('unaccent', '℗');

SELECT ts_lexize('unaccent', 'foobar');
SELECT ts_lexize('unaccent', 'ёлка');
SELECT ts_lexize('unaccent', 'ЁЖИК');
SELECT ts_lexize('unaccent', '˃˖˗˜');
SELECT ts_lexize('unaccent', 'À');
SELECT ts_lexize('unaccent', '℃℉');
SELECT ts_lexize('unaccent', '℗');

-- Controversial case.  Black-Letter Capital H (U+210C) is translated by
-- Latin-ASCII.xml as 'x', but it should be 'H'.
SELECT unaccent('ℌ');
