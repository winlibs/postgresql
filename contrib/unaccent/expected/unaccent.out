CREATE EXTENSION unaccent;
-- must have a UTF8 database
SELECT getdatabaseencoding();
 getdatabaseencoding 
---------------------
 UTF8
(1 row)

SET client_encoding TO 'UTF8';
SELECT unaccent('foobar');
 unaccent 
----------
 foobar
(1 row)

SELECT unaccent('ёлка');
 unaccent 
----------
 елка
(1 row)

SELECT unaccent('ЁЖИК');
 unaccent 
----------
 ЕЖИК
(1 row)

SELECT unaccent('˃˖˗˜');
 unaccent 
----------
 >+-~
(1 row)

SELECT unaccent('À');  -- Remove combining diacritical 0x0300
 unaccent 
----------
 A
(1 row)

SELECT unaccent('℃℉'); -- degree signs
 unaccent 
----------
 °C°F
(1 row)

SELECT unaccent('℗'); -- sound recording copyright
 unaccent 
----------
 (P)
(1 row)

SELECT unaccent('unaccent', 'foobar');
 unaccent 
----------
 foobar
(1 row)

SELECT unaccent('unaccent', 'ёлка');
 unaccent 
----------
 елка
(1 row)

SELECT unaccent('unaccent', 'ЁЖИК');
 unaccent 
----------
 ЕЖИК
(1 row)

SELECT unaccent('unaccent', '˃˖˗˜');
 unaccent 
----------
 >+-~
(1 row)

SELECT unaccent('unaccent', 'À');
 unaccent 
----------
 A
(1 row)

SELECT unaccent('unaccent', '℃℉');
 unaccent 
----------
 °C°F
(1 row)

SELECT unaccent('unaccent', '℗');
 unaccent 
----------
 (P)
(1 row)

SELECT ts_lexize('unaccent', 'foobar');
 ts_lexize 
-----------
 
(1 row)

SELECT ts_lexize('unaccent', 'ёлка');
 ts_lexize 
-----------
 {елка}
(1 row)

SELECT ts_lexize('unaccent', 'ЁЖИК');
 ts_lexize 
-----------
 {ЕЖИК}
(1 row)

SELECT ts_lexize('unaccent', '˃˖˗˜');
 ts_lexize 
-----------
 {>+-~}
(1 row)

SELECT ts_lexize('unaccent', 'À');
 ts_lexize 
-----------
 {A}
(1 row)

SELECT ts_lexize('unaccent', '℃℉');
 ts_lexize 
-----------
 {°C°F}
(1 row)

SELECT ts_lexize('unaccent', '℗');
 ts_lexize 
-----------
 {(P)}
(1 row)

-- Controversial case.  Black-Letter Capital H (U+210C) is translated by
-- Latin-ASCII.xml as 'x', but it should be 'H'.
SELECT unaccent('ℌ');
 unaccent 
----------
 x
(1 row)

